website/static/js/waterbutler.js

```diff
@@ -24,65 +21,57 @@ function getDefaultOptions(path, provider) {
+    var baseUrl = window.contextVars.waterbutlerURL + 'v1/resources/' + nid + '/providers/' + provider + path + suffix;
+
+    return baseUrl + $.param($.extend(getDefaultOptions(), options));
 }
 
-var buildCrudUrl = buildUrl.bind(this, 'file?');
-var buildCopyUrl = buildUrl.bind(this, 'copy?');
-var buildMoveUrl = buildUrl.bind(this, 'move?');
-var buildMetadataUrl = buildUrl.bind(this, 'data?');
+var buildMetadataUrl = buildUrl.bind(this, '?meta=');
+// buildRevisionsUrl needs to get v1 treatment
 var buildRevisionsUrl = buildUrl.bind(this, 'revisions?');
-var buildCreateFolderUrl = buildUrl.bind(this, 'folders?');
-
-
-function buildUploadUrl(path, provider, nid, file, options) {
-    path = (path || '/') + file.name;
-    return buildUrl('file?', path, provider, nid, options);
-}
+var buildCreateFolderUrl = buildUrl.bind(this, '?kind=folder&');
+var buildUploadUrl = buildUrl.bind(this, '?kind=file&');
 
 function buildFromTreebeard(suffix, item, options) {
-    return buildUrl(suffix, item.data.path, item.data.provider, item.data.nodeId, options);
-}
-
-function buildFromTreebeardFile(item, file, options) {
-    return buildUploadUrl(item.data.path, item.data.provider, item.data.nodeId, file, options);
+    var path
+    if ('attributes' in item.data) {
+        path = item.data.attributes.path;
+    } else {
+        path = item.data.path;
+    }
+    return buildUrl(suffix, path, item.data.provider, item.data.nodeId, options);
 }
 
-function toJsonBlob(item, options) {
-    return $.extend(getDefaultOptions(item.data.path || '/', item.data.provider), {nid: item.data.nodeId}, options);
+function buildTreeBeardUpload(item, options) {
+    var path
+    if ('attributes' in item.data) {
+        path = item.data.attributes.path;
+    } else {
+        path = item.data.path;
+    }
+    return buildUploadUrl(path, item.data.provider, item.data.nodeId, options);
 }
 
 module.exports = {
-    toJsonBlob: toJsonBlob,
-    buildDeleteUrl: buildCrudUrl,
     buildUploadUrl: buildUploadUrl,
-    buildDownloadUrl: function(path, provider, nid, options) {
-        if (window.contextVars.accessToken) {
-            options = $.extend(options || {}, {token: window.contextVars.accessToken});
-        }
-        return buildCrudUrl(path, provider, nid, options);
-    },
     buildMetadataUrl: buildMetadataUrl,
-    buildCreateFolderUrl: buildCrudUrl,
+    buildCreateFolderUrl: buildCreateFolderUrl,
     buildRevisionsUrl: buildRevisionsUrl,
-    buildTreeBeardUpload: buildFromTreebeardFile,
-    buildTreeBeardCopy: buildFromTreebeard.bind(this, 'copy?'),
-    buildTreeBeardMove: buildFromTreebeard.bind(this, 'move?'),
-    buildTreeBeardDelete: buildFromTreebeard.bind(this, 'file?'),
+    buildTreeBeardUpload: buildTreeBeardUpload,
+    buildTreeBeardFileOp: buildFromTreebeard.bind(this, ''),
+    // TODO would be nice to prompt for root delete
+    buildTreeBeardDelete: buildFromTreebeard.bind(this, '?confirm_delete=1'),
```

Comment:
Since Fangorn doesn't have an interface for storage root deletes, I would leave this out.  Right now a delete against the storage root is probably a programmer error and throwing an error is good.